subiquity/server/controllers/filesystem.py

```diff
@@ -336,20 +337,29 @@ async def _unmount_systems_dir(self):
             self._source_handler.cleanup()
             self._source_handler = None
 
-    async def _get_system(self, variation_name, label):
-        try:
-            await self._mount_systems_dir(variation_name)
-        except NoSnapdSystemsOnSource:
-            return None
-        try:
-            system = await self.app.snapdapi.v2.systems[label].GET()
-        except requests.exceptions.HTTPError as http_err:
-            log.warning("v2/systems/%s returned %s", label, http_err.response.text)
-            raise
-        finally:
-            await self._unmount_systems_dir()
-        log.debug("got system %s", system)
-        return system
+    async def _get_system(
+        self, variation_name, label
+    ) -> Tuple[Optional[SystemDetails], bool]:
```

Comment:
I'd like a docstring or similar clarifying the intent of the return value, please

Reply:
Added a docstring. Would prefer to do it in a more type-y way but that just seems verbose.